osg-koji

```diff
@@ -313,11 +313,44 @@ def verify_koji_dir(config_dir):
             (CLIENT_CERT_FILE, "User certificate file (client cert)"),
             (SERVER_CERT_FILE, "Server certificate file (CA bundle)")]:
 
+        fullpath = os.path.join(config_dir, path)
         # Note: use lexists because if we're using grid proxies then the
         # CLIENT_CERT_FILE may be a broken symlink--and that's ok
-        if not os.path.lexists(os.path.join(config_dir, path)):
+        if not os.path.lexists(fullpath):
             raise Error("%s (%s) not found in config directory (%s).\n%s"
                         % (descr, path, config_dir, RUN_SETUP_MSG))
+        # Catch broken client cert symlink separately
+        if path == CLIENT_CERT_FILE:
```

Comment:
Random thought, but it seems like this `if` block could be simplified a lot by tweaking the condition slightly:

```
if path == CLIENT_CERT_FILE and not os.path.exists(fullpath):
    target = os.readlink(fullpath)
    print("%s -> %s is a broken symlink.\n"
          "You may need to run grid-proxy-init before doing authenticated operations."
          % (fullpath, target), file=sys.stderr)
```

(if the code reaches this point, you know that `lexists` is true and `exists` is false, therefore it's safe enough to assume `fullpath` is a symlink.)

But feel free to leave it as is if you like it how you have it.

Reply:
Since it's only "ok" for the client cert to be a broken symlink, I did a bit of rearranging and it should be simpler now.